lib/Conversion/StandardToHandshake/StandardToHandshake.cpp

```diff
@@ -1391,24 +1422,7 @@ struct HandshakeRemoveBlockPass
     : public PassWrapper<HandshakeRemoveBlockPass,
                          OperationPass<handshake::FuncOp>> {
   void runOnOperation() override {
-    auto funcOp = getOperation();
-    auto &entryBlock = funcOp.getBody().front().getOperations();
-
-    // Erase all TerminatorOp, and move ReturnOp to the end of entry block.
-    for (auto &block : funcOp) {
-      Operation &termOp = block.back();
-      if (isa<handshake::TerminatorOp>(termOp))
-        termOp.erase();
-      else if (isa<handshake::ReturnOp>(termOp))
-        entryBlock.splice(entryBlock.end(), block.getOperations(), termOp);
-    }
-
-    // Move all operations to entry block and erase other blocks.
-    for (auto &block :
-         llvm::make_early_inc_range(llvm::drop_begin(funcOp, 1))) {
-      entryBlock.splice(--entryBlock.end(), block.getOperations());
-      block.erase();
-    }
+    removeBasicBlocks(getOperation());
   }
 };
```

Comment:
Maybe it's reasonable to delete the -remove-block-structure pass now?

Reply:
Maybe. I think we should keep it for the moment.  I'd like to think that we get multi-block graph regions supported before too long.